src/main/java/net/atos/zac/authentication/UserPrincipalFilter.java

```diff
@@ -70,12 +89,25 @@ public void destroy() {
 
     private LoggedInUser createLoggedInUser(final OidcSecurityContext context) {
         final IDToken idToken = context.getIDToken();
+        final Set<String> roles = Set.copyOf(context.getToken().getRealmAccessClaim().getRoles());
         return new LoggedInUser(idToken.getPreferredUsername(),
                                 idToken.getGivenName(),
                                 idToken.getFamilyName(),
                                 idToken.getName(),
                                 idToken.getEmail(),
-                                idToken.getStringListClaimValue(GROUP_MEMBERSHIP_CLAIM_NAME),
-                                context.getToken().getRealmAccessClaim().getRoles());
+                                roles,
+                                getGeautoriseerdeZaaktypen(roles));
+    }
+
+    private Set<String> getGeautoriseerdeZaaktypen(final Set<String> roles) {
+        if (roles.contains(ROL_DOMEIN_ELK_ZAAKTYPE)) {
+            return null;
```

Comment:
IK begrijp dat je hier weer gebruik maakt van het verschil tussen een null (alles) en een lege set (niets). Heb ik je vaker zien doen, en is op zich mooi. Maar in DIT geval zou ik het verschil iets explicieter maken, zodat er geen misverstanden kunnen ontstaan. Al was het maar door het toevoegen van methods isZaaktypenAlles en is ZaaktypenNiets en die consequent gebruiken om te testen waar je dat nodig hebt.